src/Components/FitnessFilters.js

```diff
@@ -50,6 +61,8 @@ class FitnessFilters extends React.Component {
     } = this.state;
     let newData = this.props.route.params.initialData;
 
+    if (selectedFavorites)
+      newData = newData.filter(drill => favoriteDrills.map(favorite => favorite.id).includes(drill.id));
```

Comment:
I think we parse all the favorite `favoriteDrills` array for each drill in `newData` maybe we could do it once before the filter instruction ? (since it is user input, it can become quite large)

Reply:
j'ai essayé qqchose, mais je ne suis pas sure que c'est ce que tu avais en tete.

Reply:
Je l'ai fait dans le 68d8feaa8109b6d41bfc34bfae064bb4f6f7ab2a